### PR TITLE
Move `Msaa` to component

### LINKED ISSUE #7194: Move Msaa to a component, instead of a resource

### PR DESCRIPTION
**Pull request by tychedelia**
Switches `Msaa` from being a globally configured resource to a per camera view component.

Closes #7194

# Objective

Allow individual views to describe their own MSAA settings. For example, when rendering to different windows or to different parts of the same view.

## Solution

Make `Msaa` a component that is required on all camera bundles.

## Testing

Ran a variety of examples to ensure that nothing broke.

TODO:
- [ ] Make sure android still works per previous comment in `extract_windows`.

---

## Migration Guide

`Msaa` is no longer configured as a global resource, and should be specified on each spawned camera if a non-default setting is desired.

### CODE REVIEW

**Review by pcwalton — APPROVED** [review]
Looks good once you fix the failing CI.

**Review by NiseVoid — APPROVED** [review]
These changes look correct to me. I always found it strange that MSAA is global, especially since having different settings on different cameras could be very useful, for example having MSAA on a camera that just sees objects with sharp outlines and using something blurrier but cheaper for the rest.

Did you test this with multiple cameras with different MSAA modes? Iirc there are some pipeline properties that depend on it, so things might break if multiple levels are enabled at once.

**Comment by tychedelia** [comment]
> Did you test this with multiple cameras with different MSAA modes? Iirc there are some pipeline properties that depend on it, so things might break if multiple levels are enabled at once.

Yes! Here are some foxes:

image
image
image

**Comment by tychedelia** [comment]
Here's a simple patch that might help alleviate the problem of having the proliferation of main textures be confusing to users who want to run split screen setups: https://github.com/bevyengine/bevy/pull/14287.

File: crates/bevy_render/src/view/mod.rs [hunk 1]
```diff
@@ -139,24 +139,27 @@ impl Plugin for ViewPlugin {
...
 ///
-/// # Example
-/// ```
-/// # use bevy_app::prelude::App;
-/// # use bevy_render::prelude::Msaa;
-/// App::new()
-///     .insert_resource(Msaa::default())
-///     .run();
+/// Note that web currently only supports 1 or 4 samples.
 /// ```
```
Comment (mockersf) [inline crates/bevy_render/src/view/mod.rs:148]:
```suggestion
```
rest of doc example has been removed

**Comment by mockersf** [comment]
the mobile example needs to be updated, or this will fail merging on android compilation

**Comment by tychedelia** [comment]
> the mobile example needs to be updated, or this will fail merging on android compilation

Should be fixed now. This PR also removes the more advanced per-window feature detection. I'd be happy to add that back in but wasn't sure where exactly it should go.

**Comment by JMS55** [comment]
I think some things were missed here. Iirc in TAA plugin docs and meshlet plugin docs, it mentions that it automatically disabled MSAA, which is no longer true.

**Pull request merged**